src/io/iter_image_recordio_2.cc

```diff
@@ -0,0 +1,532 @@
+                normalize_param_.mean_b > 0.0f || normalize_param_.mean_a > 0.0f) {
+              // subtract mean per channel
+              RGBA[0] -= normalize_param_.mean_r;
+              if (n_channels >= 3) {
+                RGBA[1] -= normalize_param_.mean_g;
+                RGBA[2] -= normalize_param_.mean_b;
+              }
+              if (n_channels == 4) {
+                RGBA[3] -= normalize_param_.mean_a;
+              }
+              for (int k = 0; k < n_channels; ++k) {
+                RGBA[k] = RGBA[k] * contrast_scaled + illumination_scaled;
+              }
+            } else if (!meanfile_ready_ || normalize_param_.mean_img.length() == 0) {
+              // do not subtract anything
+              for (int k = 0; k < n_channels; ++k) {
+                RGBA[k] = RGBA[k] * normalize_param_.scale;
+              }
+            } else {
+              CHECK(meanfile_ready_);
+              for (int k = 0; k < n_channels; ++k) {
+                  RGBA[k] = (RGBA[k] - meanimg_[k][i][j]) * contrast_scaled + illumination_scaled;
+              }
+            }
+          }
+          for (int k = 0; k < n_channels; ++k) {
+            if (!std::is_same<DType, uint8_t>::value) {
+              // normalize/mirror here to avoid memory copies
+              // logic from iter_normalize.h, function SetOutImg
+              if (is_mirrored) {
+                data[k][i][res.cols - j - 1] = RGBA[k];
+              } else {
+                data[k][i][j] = RGBA[k];
+              }
+            } else {
+              // do not do normalization in Uint8 reader
+              data[k][i][j] = RGBA[k];
+            }
+          }
+          im_data += n_channels;
+        }
+      }
+
+      mshadow::Tensor<cpu, 1> label = out.label().Back();
+      if (label_map_ != nullptr) {
+        mshadow::Copy(label, label_map_->Find(rec.image_index()));
+      } else if (rec.label != NULL) {
+        CHECK_EQ(param_.label_width, rec.num_label)
+          << "rec file provide " << rec.num_label << "-dimensional label "
+             "but label_width is set to " << param_.label_width;
+        mshadow::Copy(label, mshadow::Tensor<cpu, 1>(rec.label,
+                                                     mshadow::Shape1(rec.num_label)));
+      } else {
+        CHECK_EQ(param_.label_width, 1)
+          << "label_width must be 1 unless an imglist is provided "
+             "or the rec file is packed with multi dimensional label";
+        label[0] = rec.header.label;
+      }
+      res.release();
+    }
+  }
+#else
+      LOG(FATAL) << "Opencv is needed for image decoding and augmenting.";
+#endif
+}
+
+// create mean image.
+// NOT IMPLEMENTED
+template<typename DType>
+inline void ImageRecordIOParser2<DType>::CreateMeanImg(void) {
+  LOG(FATAL) << "Cannot find " << normalize_param_.mean_img
+             << ": creating mean image from data is not yet supported";
```

Comment:
This would require a  completely separate logic to load all the images for that (since the normalization is done in the same step as parsing), for a purely convenience method. One way to deal with it would be to have dtype version use uint8 version, which does not do normalization, as a loader.